app/frontend/utils/formio-component-traversal.ts

```diff
@@ -156,3 +157,41 @@ export const combineDiff = (formJson: IFormJson, diff: ITemplateVersionDiff) =>
   })
   return formJson
 }
+
+export const combineRevisionButtons = (formJson: IFormJson) => {
+  const convertToRevisionButton = (requirement: IFormIORequirement) => {
+    return {
+      id: requirement.id + "-revision-button",
+      key: requirement.key + "-revision-button",
+      type: "button",
+      label: "",
+      title: "Revision Button",
+      input: true,
+      action: "custom",
+      custom: `console.log("${requirement.key} says hello")`,
+      customClass: "revision-button",
+      hideLabel: true,
+      customConditional: requirement.customConditional,
+      conditional: requirement.conditional,
+    } as IFormIORequirement
+  }
```

Comment:
I don't see a need for this to be in side the parent function as it's not referencing anything internally. Place it outside so that a new function is not created every time the parent function is called

Reply:
good idea